scout/models/case/case_loading_models.py

```diff
@@ -0,0 +1,509 @@
+
+    @model_validator(mode="before")
+    def convert_cancer_int_values_to_str(cls, values) -> "SampleLoader":
+        """Sets 'msi' and 'msi' values for cancer cases to string. This is a required step in Pydantic2, in Pydantic1 values were just coerced from int to str."""
+        for item in ["msi", "tmb"]:
+            if values.get(item):
+                values[item] = str(values[item])
+        return values
+
+    @field_validator("tumor_purity", mode="before")
+    @classmethod
+    def set_tumor_purity(cls, tumor_purity: Union[str, float]) -> float:
+        """Set tumor purity value as a fraction."""
+        if isinstance(tumor_purity, str):
+            return float(Fraction(tumor_purity))
+        return tumor_purity
+
+    @model_validator(mode="before")
+    def set_sample_display_name(cls, values) -> "SampleLoader":
+        """Make sure a sample as a display name."""
+        values["display_name"] = values.get(
+            "display_name", values.get("sample_name", values.get("individual_id"))
+        )
+        return values
+
+    @model_validator(mode="before")
+    def set_alignment_path(cls, values) -> "SampleLoader":
+        """prefer key 'bam file over 'alignment_path' or 'bam_path'."""
+        values["bam_file"] = values.get(
+            "alignment_path", values.get("bam_file", values.get("bam_path"))
+        )
+        return values
+
+    @model_validator(mode="before")
+    def validate_file_path(cls, values: Dict) -> "SampleLoader":
+        """Make sure that files associated to samples (mostly alignment files) exist on disk."""
+        for item in SAMPLES_FILE_PATH_CHECKS:
+            item_path: str = values.get(item)
+            if item_path and _is_string_path(values[item]) is False:
+                raise ValueError(f"{item} path '{values[item]}' is not valid.")
+
+        return values
+
+    @field_validator("capture_kits", mode="before")
+    @classmethod
+    def set_capture_kits(cls, capture_kits: Optional[Union[str, List[str]]]) -> Optional[List]:
+        """Format capture kit (str) as a list of items."""
+        if isinstance(capture_kits, str):
+            return [capture_kits]
+        return capture_kits
+
+
+#### Custom images - related classes ####
+
+
+class Image(BaseModel):
+    """A class representing an image either associated to a case or a str variant."""
+
+    data: Optional[bytes] = None
+    description: Optional[str] = None
+    height: Optional[int] = None
+    format: Optional[str] = None
+    path: Optional[str] = None
+    str_repid: Optional[str] = None
+    title: Optional[str] = None
+    width: Optional[int] = None
+
+    @field_validator("path", mode="before")
+    @classmethod
+    def check_image_format(cls, path: str) -> Optional[str]:
+        """Make sure that the image is has standard format."""
+        image_format: str = path.split(".")[-1]
+        if image_format not in SUPPORTED_IMAGE_FORMATS:
+            raise TypeError(
```

Comment:
Checking image format is good, checking file ending a little less so. But ahem, we had this before and this is better written! Nice!